File: webapp/src/main/java/com/miaskor/todo_spring/connectors/rest/api/TaskConnector.java
package com.miaskor.todo_spring.connectors.rest.api;

import by.miaskor.domain.rest.api.TaskRequestDto;
import by.miaskor.domain.rest.api.TaskResponseDto;
import java.time.LocalDate;
import java.util.List;
import java.util.Map;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;

@FeignClient(value = "rest-api-task", url = "${connector.rest-api.url}")
public interface TaskConnector {

  @RequestMapping(value = "/tasks", method = RequestMethod.POST)
  TaskResponseDto create(@RequestBody TaskRequestDto task);

  @RequestMapping(value = "/tasks/range", method = RequestMethod.GET)
  Map<LocalDate, List<TaskResponseDto>> getAllByClientIdAndDateBetween(
      @RequestParam("date_from") String dateFrom,
      @RequestParam("date_to") String dateTo,
      @RequestParam("client_id") Integer clientId);

  @RequestMapping(value = "/tasks/date", method = RequestMethod.GET)
  List<TaskResponseDto> getAllByClientIdAndDate(
      @RequestParam String date,
      @RequestParam("client_id") Integer clientId);

  @RequestMapping(value = "/tasks/{id}", method = RequestMethod.PATCH)
  TaskResponseDto update(@PathVariable("id") Integer taskId, @RequestBody TaskRequestDto task);

  @RequestMapping(value = "/tasks/{id}", method = RequestMethod.DELETE)
  void delete(@PathVariable("id") Integer taskId);
}

File: rest-api/rest-connector/src/main/kotlin/by/miaskor/rest/connector/TaskConnector.kt
package by.miaskor.rest.connector

import by.miaskor.rest.dto.TaskDtoRequest
import by.miaskor.rest.dto.TaskDtoResponse
import feign.Headers
import feign.Param
import feign.RequestLine
import java.time.LocalDate

interface TaskConnector {

  @RequestLine("POST")
  @Headers("Content-type: application/json")
  fun create(task: TaskDtoRequest): TaskDtoResponse

  @RequestLine("GET /range?date_from={date_from}&date_to={date_to}&client_id={client_id}")
  @Headers("Content-type: application/json")
  fun getAllByClientIdAndDateBetween(
    @Param("date_from") dateFrom: String,
    @Param("date_to") dateTo: String,
    @Param("client_id") clientId: Int
  ): Map<LocalDate, List<TaskDtoResponse>>

  @RequestLine("GET /date?date={date}&client_id{client_id}")
  @Headers("Content-type: application/json")
  fun getAllByClientIdAndDate(
    @Param("date") date: String,
    @Param("client_id") clientId: Int
  ): List<TaskDtoResponse>

  @RequestLine("PATCH /{id}")
  @Headers("Content-type: application/json")
  fun update(@Param("id") taskId: Int, task: TaskDtoRequest): TaskDtoResponse

  @RequestLine("DELETE /{id}")
  @Headers("Content-type: application/json")
  fun delete(@Param("id") taskId: Int)
}
